File: webtruyen/app/Enums/UserGenderEnum.php
<?php declare(strict_types=1);

namespace App\Enums;

use BenSampo\Enum\Enum;


final class UserGenderEnum extends Enum
{
    public const FEMALE = 0;
    public const MALE = 1;
    public const LGBT = 2;
    public const SECRET = 3;


    public static function getArrayView() {
        return [
            'Nữ' => self::FEMALE,
            'Nam' => self::MALE,
            'LGBT' => self::LGBT,
            'Bí mật' => self::SECRET,
        ];
    }
    public static function getNameByValue($value) {
        return array_search($value, self::getArrayView(), true);
    }
}

File: apptruyen2/app/src/main/java/com/example/apptruyen2/constant/UserGenderEnum.kt
package com.example.apptruyen2.constant

enum class UserGenderEnum(val value: Int) {
  FEMALE(0),
  MALE(1),
  LGBT(2),
  SECRET(3);

  companion object {
    fun fromValue(value: String): UserGenderEnum {
      return when (value) {
        "Nữ" -> FEMALE
        "Nam" -> MALE
        "LGBT" -> LGBT
        else -> SECRET
      }
    }
  }
}